spec/controllers/submissions_controller_spec.rb

```diff
@@ -98,6 +108,85 @@
       end
     end
 
+    describe 'GET #finalize' do
+      let(:perform_request) { proc { get :finalize, params: {id: submission.id} } }
+      let(:cause) { 'assess' }
+
+      context 'when the request is performed' do
+        before { perform_request.call }
+
+        expect_assigns(submission: :submission)
+        expect_redirect
+      end
+
+      it 'updates cause to submit' do
+        expect { perform_request.call && submission.reload }.to change(submission, :cause).from('assess').to('submit')
+      end
+
+      context 'when contributing to a community solution is possible' do
+        let!(:community_solution) { CommunitySolution.create(exercise:) }
+
+        before do
+          allow(Java21Study).to receive(:allow_redirect_to_community_solution?).and_return(true)
+          perform_request.call
+        end
+
+        expect_redirect { edit_community_solution_path(community_solution, lock_id: CommunitySolutionLock.last) }
+      end
+
+      context 'when sharing exercise feedback is desired' do
+        before do
+          uef&.save!
+          allow_any_instance_of(Submission).to receive(:redirect_to_feedback?).and_return(true)
+          perform_request.call
+        end
+
+        context 'without any previous feedback' do
+          let(:uef) { nil }
+
+          expect_redirect { new_user_exercise_feedback_path(user_exercise_feedback: {exercise_id: submission.exercise.id}) }
+        end
+
+        context 'with a previous feedback for the same exercise' do
+          let(:uef) { create(:user_exercise_feedback, exercise:, user: current_user) }
+
+          expect_redirect { edit_user_exercise_feedback_path(uef) }
+        end
+      end
+
+      context 'with an RfC' do
+        before do
+          rfc.save!
+          allow_any_instance_of(Submission).to receive(:redirect_to_feedback?).and_return(false)
+          perform_request.call
+        end
+
+        context 'when an own RfC is unsolved' do
+          let(:rfc) { create(:rfc, user: current_user, exercise:, submission:) }
+          let(:flash_message) { I18n.t('exercises.editor.exercise_finished_redirect_to_own_rfc') }
```

Comment:
This variable is not used so the line probably could be removed. For me locally the tests pass without the line.